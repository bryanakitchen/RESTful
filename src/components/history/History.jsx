import React from 'react';
import HistoryItem from './HistoryItem';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const HistoryDiv = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 30%;
  background: #0A1128;
  border-radius: 5px;
  margin: 0px 5px 0px 10px;
`;

const HistoryHeader = styled.p`
  text-align: center;
  font-size: 20px;
  margin: 6px 3px;
  color: #E9ECEF;
`;

const HistoryBody = styled.span`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  color: #001F54;
  margin: 5px;
  `;
  
const ListItem = styled.span`
  word-wrap: break-word;
  padding: .3rem;
  margin: 3px;
  max-width: 90%;
  border: 1px solid #034078;
  border-radius: 8px;
  background: #E9ECEF;


`;

const History = ({ history }) => {
  const historyElements = history.map((item, i) => {
    return (
      <ListItem key={i}>
        <HistoryItem method={item.method} url={item.url} id={i} />
      </ListItem>
    );
  });

  return (
    <>
      { (historyElements < 1) 
        ? ' '
        : 
        <HistoryDiv>
          <HistoryHeader>History</HistoryHeader>
          <HistoryBody>
            {historyElements}
          </HistoryBody>        
        </HistoryDiv>
      }
    </>
  );
};

History.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    method: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  }))
};

export default History;
