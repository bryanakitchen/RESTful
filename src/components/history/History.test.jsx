import React from 'react';
import { render, cleanup } from '@testing-library/react';
import History from './History';

describe('History component', () => {
  afterEach(() => cleanup());
  it('renders History', () => {
    const { asFragment } = render(<History
      history={[
        {
          method: 'get',
          url: 'google.com',
          id: 0
        },
        {
          method: 'put',
          url: 'google1.com',
          id: 1
        }
      ]}
    />);    
      
    expect(asFragment()).toMatchSnapshot();
  });
});
