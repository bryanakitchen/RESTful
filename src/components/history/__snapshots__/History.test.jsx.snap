// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History component renders History 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm htbDCf"
  >
    <p
      class="sc-hKgILt kJfJMY"
    >
      History
    </p>
    <span
      class="sc-eCssSg hUWzSP"
    >
      <span
        class="sc-jSgupP ffeFYP"
      >
        <article
          id="0"
        >
          <span
            class="sc-bdfBwQ gdFaFC"
            id="0"
          >
            get
          </span>
          <p
            class="sc-gsTCUz khlkDi"
            id="0"
          >
            google.com
          </p>
        </article>
      </span>
      <span
        class="sc-jSgupP ffeFYP"
      >
        <article
          id="1"
        >
          <span
            class="sc-bdfBwQ gdFaFC"
            id="1"
          >
            put
          </span>
          <p
            class="sc-gsTCUz khlkDi"
            id="1"
          >
            google1.com
          </p>
        </article>
      </span>
    </span>
  </div>
</DocumentFragment>
`;
